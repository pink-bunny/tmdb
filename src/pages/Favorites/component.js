import React from 'react';
import PropTypes from 'prop-types';
import {
  Layout, Row, Col, Typography
} from 'antd';
import List from '../../components/List';

const Favorites = ({
  list,
  loading,
  error,
  totalItems,
  currentPage,
  fetchList,
  removeModalAction
}) => (
  <Layout>
    <Layout.Content>
      <Row>
        <Col
          offset={2}
          span={20}
        >
          <div className="top-margin">
            <Typography.Title>
              Favorites
            </Typography.Title>
          </div>
        </Col>
      </Row>

      <List
        emptyListTxt="No movies in favorites found"
        list={list}
        loading={loading}
        error={error}
        totalItems={totalItems}
        currentPage={currentPage}
        fetchList={fetchList}
        removeModalAction={removeModalAction}
        removeModalTxt="Do you want to delete this item from the favorites?"
      />

    </Layout.Content>
  </Layout>
);

Favorites.propTypes = {
  fetchList: PropTypes.func.isRequired,
  removeModalAction: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string.isRequired,
  list: PropTypes.arrayOf(PropTypes.object),
  totalItems: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired
};
Favorites.defaultProps = {
  list: null
};

export default Favorites;
