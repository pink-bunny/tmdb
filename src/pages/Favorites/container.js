import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import FavoritesComponent from './component';
import {
  fetchFavorites as fetchFavoritesAction,
  toggleToFavorite as toggleToFavoriteAction
} from '../../../state/favorites/actions';
import {
  isFavoritesLoading,
  favoritesTotalItems,
  favoritesCurrentPage,
  favoritesError,
  favoritesList
} from '../../../state/favorites/selectors';

const FavoritesContainer = ({
  list,
  loading,
  error,
  totalItems,
  currentPage,
  fetchFavorites,
  toggleToFavorite
}) => (
  <FavoritesComponent
    list={list}
    loading={loading}
    error={error}
    totalItems={totalItems}
    currentPage={currentPage}
    fetchList={fetchFavorites}
    removeModalAction={toggleToFavorite}
  />
);

FavoritesContainer.propTypes = {
  fetchFavorites: PropTypes.func.isRequired,
  toggleToFavorite: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string.isRequired,
  list: PropTypes.arrayOf(PropTypes.object),
  totalItems: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired
};
FavoritesContainer.defaultProps = {
  list: null
};

const mapDispatchToPtops = {
  fetchFavorites: fetchFavoritesAction,
  toggleToFavorite: toggleToFavoriteAction
};

const mapStateToProps = (state) => ({
  loading: isFavoritesLoading(state),
  list: favoritesList(state),
  error: favoritesError(state),
  totalItems: favoritesTotalItems(state),
  currentPage: favoritesCurrentPage(state)
});

export default connect(mapStateToProps, mapDispatchToPtops)(FavoritesContainer);
