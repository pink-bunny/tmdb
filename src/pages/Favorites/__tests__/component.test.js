import React from 'react';
import { shallow } from 'enzyme';

import Favorites from '../component';

describe('Favorites component', () => {
  const requiredProps = {
    loading: false,
    error: '',
    totalItems: 21,
    currentPage: 2,
    fetchList: jest.fn(),
    removeModalAction: jest.fn()
  };
  const wrapper = shallow(<Favorites {...requiredProps} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */

  it('matches snapshot with required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot with optional props', () => {
    wrapper.setProps({
      list: [{ id: 1, name: 'Mock name' }]
    });
    expect(wrapper).toMatchSnapshot();
  });
});
