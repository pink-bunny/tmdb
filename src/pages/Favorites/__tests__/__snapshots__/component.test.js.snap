// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component matches snapshot with optional props 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            Favorites
          </Title>
        </div>
      </Col>
    </Row>
    <ListContainer
      currentPage={2}
      emptyListTxt="No movies in favorites found"
      error=""
      fetchList={[MockFunction]}
      list={
        Array [
          Object {
            "id": 1,
            "name": "Mock name",
          },
        ]
      }
      loading={false}
      removeModalAction={[MockFunction]}
      removeModalTxt="Do you want to delete this item from the favorites?"
      totalItems={21}
    />
  </Adapter>
</Adapter>
`;

exports[`Favorites component matches snapshot with required props 1`] = `
<Adapter>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        offset={2}
        span={20}
      >
        <div
          className="top-margin"
        >
          <Title>
            Favorites
          </Title>
        </div>
      </Col>
    </Row>
    <ListContainer
      currentPage={2}
      emptyListTxt="No movies in favorites found"
      error=""
      fetchList={[MockFunction]}
      list={null}
      loading={false}
      removeModalAction={[MockFunction]}
      removeModalTxt="Do you want to delete this item from the favorites?"
      totalItems={21}
    />
  </Adapter>
</Adapter>
`;
